packages/create-turbo/src/utils/examples.ts

```diff
@@ -0,0 +1,135 @@
+import got from "got";
+import tar from "tar";
+import { Stream } from "stream";
+import { promisify } from "util";
+import { join } from "path";
+import { tmpdir } from "os";
+import { createWriteStream, promises as fs } from "fs";
+
+const pipeline = promisify(Stream.pipeline);
+
+export type RepoInfo = {
+  username: string;
+  name: string;
+  branch: string;
+  filePath: string;
+};
+
+export async function isUrlOk(url: string): Promise<boolean> {
+  try {
+    const res = await got.head(url);
+    return res.statusCode === 200;
+  } catch (err) {
+    return false;
+  }
+}
+
+export async function getRepoInfo(
```

Comment:
The github url parsing logic here seems like something we can find from an existing package 🤔 . I've personally implemented this so many times in so many languages and I always feel like it's an easy place to save some intenral LOC. It's not a lot of code and sunk cost etc etc, but just a suggestion

Reply:
This is also from create-next-app

Reply:
can we publish it as a library and use it instead of copy pasting?

Reply:
Yep, i'm planning to talk to the next team about this to see about a core-create or similar we could all utilize and open source.